RFPStrategy/RFPRecipient/spec.ts

```diff
@@ -0,0 +1,88 @@
+import { Address, BeforeAll, BigInt, Event, LiveObject, OnEvent, Property, Spec } from '@spec.dev/core'
+
+import { decodeRfpRegistrationData } from '../../shared/decode'
+import { getStatusFromInt } from '../../shared/status'
+
+/**
+ * RFP details
+ */
+@Spec({
+    uniqueBy: ['strategyId', 'id', 'chainId']
+})
+class RFPRecipient extends LiveObject {
+
+    @Property()
+    recipientId: Address
+
+    @Property()
+    strategyId: Address
+
+    @Property({ default: 0 })
+    proposalBid: BigInt
+
+    @Property()
+    isRegistryAnchor: boolean
+
+    @Property()
+    status: string
+
+    @Property()
+    metadataProtocol: number
+
+    @Property()
+    metadataPointer: string
+
+    @Property()
+    sender: Address
+
+    // ====================
+    // =  Event Handlers  =
+    // ====================
+
+    @BeforeAll()
+    setCommonProperties(event: Event) {
+        this.strategyId = event.origin.contractAddress;
+        this.recipientId = event.data.recipientId;
+    }
+
+    @OnEvent('allov2.RFPSimple.Registered')
+    async onRegistration(event: Event) {
```

Comment:
Since we're only upserting a single record in this handler, we can actually just use `this` as the model representation of the record that will get auto-saved. And since both `this.strategyId` and `this.recipientId` are already set for us in the `@BeforeAll` handler above, this one can just be:
```typescript
@OnEvent('allov2.RFPSimple.Registered')
async onRegistration(event: Event) {
    const useRegistryAnchor = await this.contract.useRegistryAnchor()

    const { proposalBid, metadata } = decodeRfpRegistrationData(
        useRegistryAnchor, 
        event.data.data,
    )

    this.proposalBid = proposalBid
    this.isRegistryAnchor = useRegistryAnchor
    this.status = getStatusFromInt(0)
    this.metadataProtocol = metadata.protocol
    this.metadataPointer = metadata.pointer
    this.sender = event.data.sender
}
```